metadata_layer.py

```diff
@@ -1176,9 +1176,11 @@ def apply(self, _db, collection, replace=None):
             if link.rel in Hyperlink.CIRCULATION_ALLOWED:
                 link_obj = link_objects[link]
                 if replace.mirror:
-                    # We need to mirror this resource. If it's an image, a
-                    # thumbnail may be provided as a side effect.
-                    self.mirror_link(pool, data_source, link, link_obj, replace)
+                    for mirror_type in replace.mirror.keys():
+                        if replace.mirror[mirror_type]:
+                            # We need to mirror this resource. If it's an image, a
+                            # thumbnail may be provided as a side effect.
+                            self.mirror_link(pool, data_source, link, link_obj, replace, mirror_type)
```

Comment:
I don't understand this piece. I feel like this is the point where you need to decide which of the mirrors in replace.mirror to use for this particular link -- so you do need new code here and looking at replace.mirror is appropriate. But I would expect it to have something to do with `link.rel` -- images go to the image mirror, books go to the book mirror. Maybe there needs to be a `ReplacementPolicy.mirror_for(LinkData)` method to sort this out.

I think the code you have is going to tell _every_ mirror to mirror every link.

Reply:
So would it make sense to assume that the link should be mirrored for cover images if the `rel` is in `MIRRORED = [OPEN_ACCESS_DOWNLOAD, IMAGE, THUMBNAIL_IMAGE]`? Otherwise, it is a book mirror. In this specific case, there is a check to see if `rel` is in `CIRCULATION_ALLOWED` so that limits what type it is and I guess in this case, it is a book mirror?

Reply:
Yes, it sounds like in this case it's a book mirror. For `[IMAGE, THUMBNAIL_IMAGE]` it would be a cover mirror.